tests/zfs-tests/tests/functional/cli_root/zpool_reopen/zpool_reopen.shlib

```diff
@@ -0,0 +1,140 @@
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2016, 2017 by Intel Corporation. All rights reserved.
+# Copyright (c) 2017 Open-E, Inc. All Rights Reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/tests/functional/cli_root/zpool_reopen/zpool_reopen.cfg
+
+# part of logic borrowed from functional/fault test
+
+function get_debug_device
+{
+	lsscsi | nawk '/scsi_debug/ {print $6; exit}' | cut -d / -f3
+}
+
+function get_scsi_host #disk
+{
+	typeset disk=$1
+	ls /sys/block/$disk/device/scsi_device | cut -d : -f 1
+}
+
+function clear_labels #disks
+{
+	for disk in $@; do
+		if ( is_loop_device $disk ) || ( is_mpath_device $disk ); then
+			zpool labelclear -f /dev/$disk
+		else
+			zpool labelclear -f /dev/${disk}1
+		fi
+	done
+}
+
+# Set the REMOVED_DISK and REMOVED_DISK_ID constants for device
+# used for re-plugging. When the disk is loop device use the
+# scsi_debug emulated drive. Otherwise use the real drive.
+function set_removed_disk
+{
+	if is_loop_device $DISK1; then
+		export REMOVED_DISK=$(get_debug_device)
+		export REMOVED_DISK_ID=$(get_persistent_disk_name $REMOVED_DISK)
+	elif ( is_real_device $DISK1 ) || ( is_mpath_device $DISK1 ); then
+		export REMOVED_DISK="$DISK1"
+		export REMOVED_DISK_ID=${devs_id[0]}
+	else
+		log_fail "No drives that supports removal"
+	fi
+}
+
+# Simulate unplug action of the given disk
+function unplug #disk
```

Comment:
How about `unplug_disk`, `plug_disk` to prevent any confusion.

Reply:
Ok.